packages/discovery-react-components/src/utils/__tests__/formatMessage.test.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { formatMessage } from '../formatMessage';
+
+describe('formatMessage util', () => {
+  test('correctly formats a message when provided with a message and elements to interpolate', () => {
+    const messageToFormat = 'Format me with a {button} and a {link}';
+    const button = <button>I am button</button>;
+    const link = <a>I am link</a>;
+    const formattedMessage = render(
+      <div>{formatMessage(messageToFormat, { button: button, link: link })}</div>
+    );
+    const buttonMatch = formattedMessage.getByText('I am button');
```

Comment:
does it work when you do `formattedMessage.getByText('Format me with a I am button and a I am link')`?

Reply:
No, since the text is broken up across elements, this doesn't work...